README.md

```diff
@@ -1,204 +1,270 @@
+own environment and, consequently, its libraries in specific versions.
+
+There are several tools that create Python _virtual environments_, one of the
+most well-known is [virtualenv](https://pypi.org/project/virtualenv/).
+This is what we are going to use in this example.
 
-As this service is written in **Python 2.7**, it is recommended that all packages are
-installed within a [virtual environment](https://github.com/pypa/virtualenv).
-It can be created by:
+We will use [pip](https://pypi.org/project/pip/) (_Package Installer for Python_)
+to install _virtualenv_, in this case, we will use _pip_ in version _19.0.3_.
 
-```shell
-virtualenv ./venv
-source ./venv/bin/activate
+Make sure to [install the pip]((https://pip.pypa.io/en/stable/installing/))
+according to your operating system, then [install virtualenv](https://virtualenv.pypa.io/en/latest/installation.html#via-pip).
+That done, we will be able to create and manage our virtual environment to run
+the service.
+
+First let's create a new _virtual environment_ using the following command:
+
+```bash
+# The most common is to create virtualenv at the
+# root of the project that it will belong to.
+$> virtualenv ./venv
 ```
 
-Which will create a `venv` folder in current directory and install
-basic Python packages there. The second command will set all environment
-variables so that new Python packages are installed there as well.
+After creating a _virtual environment_, we need to activate it so that we can
+install the necessary project packages. For this, we use the following command:
 
-To install all dependencies from Auth, execute the following commands.
+```bash
+$> source ./venv/bin/activate
+```
 
-```shell
-# you may need sudo for those
-apt-get install -y python-pip versiontools
-# There is a package that is not in PyPI
-pip install -r ./requirements/requirements.txt
-python setup.py install
+__NOTE THAT__ to exit the _virtual environment_ we use the `deactivate` command,
+as we can see below:
+
+```bash
+# When you no longer want to run the History service,
+# you need to leave the virtual environment created for it:
+$> deactivate ./venv
 ```
 
-The last command will generate a .egg file and install it into your virtual
-environment.
+To install all dependencies, execute the following commands.
 
-#### Docker Setup
+```bash
+# Some libraries depend on the specific version of the pip:
+$> python -m pip install pip==19.0.3
 
-Another alternative is to use **docker** to run the service. To build the
-container, from the repository's root:
+$> pip install versiontools
 
-```shell
+$> pip install -r ./requirements/requirements.txt
+```
+
+The next and last command will generate a
+[`.egg`](http://peak.telecommunity.com/DevCenter/PythonEggs) file and install
+it into your virtual environment:
+~~~bash
+$> python setup.py install
+~~~
+
+#### **History service on Docker**
+
+Another alternative is to use [Docker](https://www.docker.com/)  to run the
```

Comment:
__NOTE THAT__  you can use the official images provided by dojot in its  [DockerHub page](https://hub.docker.com/r/dojot/).